src/modules/projectSearchContext.js

```diff
@@ -0,0 +1,35 @@
+class ProjectSearchContext {
+
+    /**
+     * Create a search context for a specific project. This can be used to search for traces.
+     * 
+     * @param {SessionManager} sessionManager The session manager, used for connections to the Hansken servers
+     * @param {UUID} projectId The project id
+     */
+    constructor(sessionManager, projectId) {
+        this.sessionManager = sessionManager;
+        this.projectId = projectId;
+    }
+
+    /**
+     * Search for traces in a project.
+     * 
+     * @param {string} projectId 
+     * @param {string|object} request The query as an HQL query string, or the full REST request for the /search
+     * @returns 
+     */
+    traces = (request = '') => {
```

Comment:
maybe call this hql?

Then it becomes project('').search().hql('').then(...)

Or maybe project('').traces('insert your HQL here')

Or project('').traces().filter('insert HQL here')

Reply:
I expect that most users will use a proper search request object and the string hql is merely a fallback for lazy people...

Reply:
Also note that there will be coming more functions to find mvt, tracelets, so I'd like to keep search traces to a single method

Reply:
ah ok